client/extensions/woocommerce/app/products/product-form-variations-modal.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+import classNames from 'classnames';
+import { localize } from 'i18n-calypso';
+import { find } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import formattedVariationName from '../../lib/formatted-variation-name';
+import FormFieldSet from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
+import FormTextArea from 'components/forms/form-textarea';
+import FormToggle from 'components/forms/form-toggle';
+import VerticalMenu from 'components/vertical-menu';
+
+class ProductFormVariationsModal extends React.Component {
+
+	static propTypes = {
+		product: PropTypes.object.isRequired,
+		variations: PropTypes.array.isRequired,
+		editProductVariation: PropTypes.func.isRequired,
+	};
+
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			selectedVariation: this.props.selectedVariation,
+		};
+
+		this.switchVariation = this.switchVariation.bind( this );
+		this.setDescription = this.setDescription.bind( this );
+		this.toggleVisible = this.toggleVisible.bind( this );
+	}
+
+	setDescription( e ) {
+		const { selectedVariation } = this.state;
+		const { product, editProductVariation, variations } = this.props;
+		const variation = find( variations, ( v ) => selectedVariation === v.id );
+		editProductVariation( product, variation, { description: e.target.value } );
+	}
+
+	toggleVisible() {
+		const { selectedVariation } = this.state;
+		const { product, editProductVariation, variations } = this.props;
+		const variation = find( variations, ( v ) => selectedVariation === v.id );
+		editProductVariation( product, variation, { visible: ! variation.visible } );
+	}
+
+	switchVariation( selectedVariation ) {
+		this.setState( {
+			selectedVariation
+		} );
+	}
+
+	render() {
+		const { variations, translate } = this.props;
+		const { selectedVariation } = this.state;
+		const variation = find( variations, ( v ) => selectedVariation === v.id );
```

Comment:
Maybe a helper function for finding the currently selected variation? You've used this code 3 places.